BaseStation/SensorDiagnostics/SensorDiagnostics/LogSensors.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Scarlet;
+using Scarlet.Components;
+using Scarlet.Components.Sensors;
+using Scarlet.Utilities;
+using System.IO;
+
+namespace FlatRobot
+{
+    static class LogSensors
+    {
+        private static List<List<string>>[] fileOutput = new List<List<string>>[7];
+        private static StreamWriter SensorLogFile;
+
+        private const string delimiter = ",";
+        private const string LogFilesLocation = "Logs";
+
+        public static void logSensors()
+        {
+            int[] sensorCount = new int[7];
+            /*
+             * 0 - Encoders
+             * 1 - LimitSwtichs
+             * 2 - MAX21855s
+             * 3 - MPU6050s
+             * 4 - Potentiometers
+             * 5 - VEML6070s
+             */
+            foreach (ISensor s in Global.sensors)
```

Comment:
Do the WriteLines in this first pass to avoid the need to store the data in an intermediate structure. (fileOutput)

Reply:
I will do fix this soon while im implementing the new datalog class from scarlet. Thanks!